unblob/finder.py

```diff
@@ -2,166 +2,202 @@
 Searching Chunk related functions.
 The main "entry point" is search_chunks_by_priority.
 """
-import io
+from enum import Flag
 from functools import lru_cache
-from operator import itemgetter
-from pathlib import Path
-from typing import Dict, List, Tuple, Type
+from typing import Dict, List, Optional, Tuple, Type
 
-import yara
+import attr
+import hyperscan
 from structlog import get_logger
 
-from .file_utils import InvalidInputFormat, LimitedStartReader
+from .file_utils import InvalidInputFormat
 from .handlers import Handlers
 from .logging import noformat
-from .models import Handler, TaskResult, ValidChunk, YaraMatchResult
+from .models import File, Handler, TaskResult, ValidChunk
 from .report import CalculateChunkExceptionReport
 
 logger = get_logger()
 
 
-_YARA_RULE_TEMPLATE = """
-rule {NAME}
-{{
-    {YARA_RULE}
-}}
-"""
+@attr.define
+class HyperscanMatchContext:
+    handler_map: Dict[int, Handler]
+    file: File
+    file_size: int
+    all_chunks: List
+    task_result: TaskResult
+
+
+class ScanContinue(Flag):
+    Continue = False
+    Terminate = True
+
+
+def _calculate_chunk(
+    handler: Handler, file: File, real_offset, task_result: TaskResult
+) -> Optional[ValidChunk]:
+    # limited_reader = LimitedStartReader(file, real_offset)
```

Comment:
Please delete unused / out-of-date stuff instead of just commenting them.

Reply:
done